miso-web/src/main/webapp/scripts/sample_hot.js

```diff
@@ -1457,5 +1483,101 @@ Sample.hot = {
       }
     }
     return (Hot.messages.failed.length ? false : true);
+  },
+  
+  /**
+   * Take the data in the external names column and send it to the server to find matching identities
+   */
+  lookupIdentities: function () {
+    var lookupButton = document.getElementById('lookupIdentities');
+    lookupButton.setAttribute('disabled', 'disabled');
+    var identitiesSearches = [];
+    var blankRows = []
+    var externalName;
+    for (var i = 0; i < Hot.startData.length; i++) {
+      externalName = Hot.startData[i].externalName;
+      if (externalName == null || externalName == undefined || externalName.length == 0) {
+        blankRows.push(i + 1);
+      } else {
+        externalName = externalName.replace(",", ";");
+        identitiesSearches.push(externalName);
+      }
+    }
+    if (blankRows.length || !Hot.startData.length) {
+      var msg = "Rows " + blankRows.join(", ") + ": External name can not be blank";
+      Hot.addErrors({ failed: [msg] });
+      lookupButton.removeAttribute('disabled');
+      return false;
+    } else {
+      Sample.hot.identityRequestCounter++;
+      jQuery.ajax({
+        url:"/miso/rest/tree/identities",
+        data: "{\"identitiesSearches\":" + JSON.stringify(identitiesSearches) 
+                 + ", \"requestCounter\":" + Sample.hot.identityRequestCounter + "}", 
+        contentType:'application/json; charset=utf8',
+        dataType: 'json',
+        type: 'POST'
+      }).complete(function (data) {
+        console.log(data);
+        lookupButton.removeAttribute('disabled');
+      }).success(function (data) {
+        // make sure there haven't been any newer requests sent while we were fetching data
+        if (data.requestCounter == Sample.hot.identityRequestCounter) {
+          Sample.hot.foundIdentities = data.foundIdentities;
+          Sample.hot.flattenedIdentities = [].concat.apply([], Sample.hot.foundIdentities);
+          Sample.hot.setIdentitySources();
+          Sample.hot.displayCheckmark();
+        }
+      });
+    }
+  },
+  
+  /**
+   * Processes returned AJAX data for identities and sets source for cells in table
+   */
+  setIdentitySources: function () {
+    var identityColIndex = Hot.getColIndex('identityAlias');
+    var rowCount = Sample.hot.foundIdentities.length;
+    var selectedProjectId = Sample.hot.selectedProjectId;
+    for (var i = 0; i < rowCount; i++) {
+      var hasIdentityInProject = false;
+      var sortedIdentities = Sample.hot.foundIdentities[i].sort(function (a, b) {
+        hasIdentityInProject = (a.projectId == selectedProjectId);
+        return (a.projectId == selectedProjectId || a.projectId > b.projectId) ? 1 : ((b.projectId > a.projectId) ? -1 : 0);
```

Comment:
The mutation of `hasIdentityInProject` from inside the comparison is not guaranteed to work. If only one item is returned, the list is sorted, but the comparison function will never be called. The comparison itself is also hard to reason about. Might I suggest:
```
var sortedIdentities = Sample.hot.foundIdentities[i].sort(function (a, b) {
  var aSortId = a.projectId == selectedProjectId ? 0 : a.projectId;
  var bSortId = b.projectId == selectedProjectId ? 0 : b.projectId;
  return aSortId - bSortId;
});
var hasIdentityInProject = (sortedIdentities.length > 0 && sortedIdentities[0].projectId != selectedProjectId);
var identityNames = sortedIdentities.map(Sample.hot.getIdentityLabel);
if (!hasIdentityInProject) sortedIdentities.unshift("First Receipt");
```
I would also avoid changing the contents of a variable from one type to another. It's better to create a new variable.